### ISSUE TITLE
Scale recorded audio with quen text to speech model

### ISSUE THREAD

**Issue by LNshuti** [LNshuti/iga-tts#5]
### Objective

Finetune or clone an open-source Qwen3-TTS model using my provided Kinyarwanda voice assets, then batch-generate all missing Kinyarwanda audio for my app. Integrate with my existing repository pipeline (IDs, paths, npm scripts) and deliver a reproducible training/inference setup using Hugging Face Skills.

### Defaults if not specified
Approach: Start with short/zero-shot voice cloning (60–120s reference), fall back to LoRA finetuning if needed for quality/consistency
Model: Qwen/Qwen3-TTS-12Hz-1.7B-Base (fallback to 0.6B if GPU-constrained)
Compute: Hugging Face Jobs A10G/A100 (single-GPU)
Language: Kinyarwanda (rw)
Output audio: 24 kHz mono WAV per phrase in corpus/kinyarwanda/{ID}.wav, then convert to Opus/MP3 via existing npm script

### Use these skills
hugging-face-cli: manage Hub repos and files
hugging-face-datasets: create/maintain dataset repo with metadata
hugging-face-jobs: run preprocessing, cloning/finetune, inference at scale
hugging-face-evaluation: record eval metrics and add to model card

### Important context
Project repo includes an audio workflow and ID mapping:
Missing phrases: npm run corpus:status --missing
WAV inputs (recorded or synthetic) live in corpus/kinyarwanda/{ID}.wav
Build converts to public/audio/kinyarwanda/{ID}.opus and {ID}.mp3 via: npm run corpus:build
Sample rate goal: 24 kHz mono, 16-bit PCM WAV
Qwen3-TTS supports multilingual TTS and 3-second voice cloning. We aim for natural Kinyarwanda pronunciation and consistent speaker style with my assets.

### High-level plan (reason before doing)
- Inspect repo structure and missing IDs; verify available voice assets and transcripts; confirm language code and sample rate constraints.
- Prepare a clean, reproducible dataset of (text, audio) pairs for the provided voice, suitable for cloning and (if needed) LoRA finetuning.
- Try short/zero-shot cloning for the target voice; do a pilot synthesis for a small Kinyarwanda set; evaluate naturalness, speaker similarity, and pronunciation.
- If quality is insufficient, run LoRA finetuning with my voice data; re-evaluate.
- Generate WAVs for all missing phrase IDs; integrate with existing build to produce Opus/MP3; commit and document.
- Publish artifacts (dataset, model/adapter, inference script, evaluation) and provide a rollback plan.

### Step-by-step instructions for the agent
Setup and repo discovery
/plugin marketplace add huggingface/skills
/plugin install hugging-face-cli@huggingface/skills
/plugin install hugging-face-datasets@huggingface/skills
/plugin install hugging-face-jobs@huggingface/skills
/plugin install hugging-face-evaluation@huggingface/skills
Locate the project repo. If not already loaded, ask for the repo URL or local path. Confirm these exist:
corpus/kinyarwanda/ (WAVs, numeric IDs)
public/audio/kinyarwanda/ (built outputs)
npm scripts (corpus:status, corpus:build)
Run: npm run corpus:status --missing and capture the list of missing phrase IDs and their Kinyarwanda text.

Dataset creation (for reproducibility and finetune fallback)
Create a Hugging Face dataset repo (private by default) named <org>/igisha-kinyarwanda-voice-<speaker>-v1
Build metadata.csv with columns: utt_id,text,language,audio_path,speaker_id,split
  Example rows:
"Muraho","rw","corpus/kinyarwanda/1.wav","teacher1","train"
"Mwaramutse","rw","corpus/kinyarwanda/2.wav","teacher1","train"
Audio prep (run on HF Jobs if needed):
Validate/convert provided voice WAVs to 24 kHz mono 16-bit PCM
Loudness normalize to around -23 LUFS and trim leading/trailing silence to <=50 ms
Ensure each clip length is 1–8s; flag outliers
Keep a train/val/test split (e.g., 90/5/5 by utterance)
Push metadata and audio (or symlinks/paths) to the dataset repo. Document normalization commands used (e.g., sox/ffmpeg filters).

C) Baseline: voice cloning without finetune
Select Qwen/Qwen3-TTS-12Hz-1.7B-Base (fallback to 0.6B if GPU VRAM insufficient)
From the provided voice assets, assemble 30–180 seconds of clean reference speech + transcripts (Kinyarwanda or any language, but prefer Kinyarwanda if available)
Follow the Qwen3-TTS README for the official cloning API. Save a reusable "voice profile" artifact (embedding/token) and doc how to invoke it (CLI/py snippet)
Pilot synthesis for ~30 diverse Kinyarwanda phrases spanning phonetic variety; export WAV 24 kHz mono
Evaluate quickly:
Speaker similarity: cosine similarity with a speaker encoder (e.g., ECAPA-TDNN)
Intelligibility/pronunciation: run whisper-small/multilingual ASR on synth audio and compute CER/WER vs source text
Subjective check: 5–10 sample audio clips
If metrics and subjective quality are acceptable, proceed to full generation.

D) Finetune fallback (LoRA on Qwen3-TTS)
- If cloning quality is insufficient, run LoRA finetuning using the dataset created in B.
- Use Hugging Face Jobs with GPU (prefer A10G 24GB+ or A100 40/80GB). Save and expose the exact training command, environment, and commit hash.
- Suggested starting hyperparameters (adjust per official Qwen3-TTS training guide):
  - Optimizer: AdamW, lr 1e-4, weight_decay 0.01
  - Batch size: effective 16–64 (grad accumulation if needed)
  - LoRA rank: 16–32; alpha: 32; dropout: 0.05
  - Freeze base model weights; train adapter layers only
  - Epochs/steps: target 10–30 minutes of voice => 3–5 epochs; early stop on val loss
- Log training with hugging-face-trackio if available; checkpoint every N steps
- Evaluate with the same pipeline as in C; select best checkpoint; push adapter and config to HF Hub (<org>/igisha-kinyarwanda-voice-<speaker>-lora-v1)

E) Batch synthesis for missing phrases
- Source missing IDs and text via: npm run corpus:status --missing
- For each missing ID, synthesize 24 kHz mono WAV using the chosen method:
  - Cloning: use the saved voice profile
  - Finetuned: load the base model + LoRA adapter
- Save to corpus/kinyarwanda/{ID}.wav (numeric IDs only to match pipeline)
- Run npm run corpus:build to produce Opus and MP3 into public/audio/kinyarwanda/
- Re-run npm run corpus:status to confirm coverage is now 100% for target categories

F) Deliverables and integration
- Links:
  - HF dataset repo URL
  - HF model/adapters repo URL
  - HF Jobs run pages (preprocessing, finetune, synthesis)
- In the app repo:
  - New WAVs in corpus/kinyarwanda/
  - Generated .opus and .mp3 in public/audio/kinyarwanda/
  - A scripts/tts/generate_kinyarwanda.py (or CLI) with instructions to resynthesize on demand
  - A docs/TTS_KINYA_README.md including: approach, data used, metrics, how to reproduce, usage examples
- Evaluation summary:
  - Speaker similarity stats
  - ASR CER/WER on a Kinyarwanda validation set
  - 5–10 audio samples with IDs

G) Safety, consent, and license notes
Confirm the speaker’s consent for cloning/finetuning and redistribution of generated audio
Add license/consent line to model card and repo docs
Qwen3-TTS license: Apache-2.0; ensure model card mentions it and any adapter license

### Examples and snippets
- Synthesis loop (pseudo-Python; adapt to official Qwen3-TTS API):
  for row in missing_rows: # row.id, row.text
      wav = tts_synthesize(text=row.text, lang='rw', voice_profile='teacher1')
      save_wav(f'corpus/kinyarwanda/{row.id}.wav', wav, sr=24000)

- FFmpeg normalization example (use in preprocessing if needed):
  ffmpeg -i in.wav -ar 24000 -ac 1 -af "loudnorm=I=-23:TP=-2:LRA=7, silenceremove=start_periods=1:start_silence=0.05:start_threshold=-40dB:stop_periods=1:stop_silence=0.05:stop_threshold=-40dB" out.wav

- Metadata.csv minimal example:
  utt_id,text,language,audio_path,speaker_id,split
  1,Muraho,rw,corpus/kinyarwanda/1.wav,teacher1,train

### Acceptance criteria
100% of targeted Kinyarwanda phrases have .wav, .opus, and .mp3 generated and correctly named by ID
Reproducible dataset and (if applicable) LoRA adapter on Hugging Face Hub with model card
Documented synthesis script and instructions to regenerate
Evaluation report with similarity metrics and an intelligibility proxy (ASR-based), plus audio samples
CI step or script that verifies coverage and file integrity

If anything above is missing or blocked (e.g., repo access, insufficient GPU), pause and ask for the missing info, proposing practical alternatives (e.g., smaller model, CPU-only cloning preview, staged generation).
